kalite/distributed/management/commands/setup.py

```diff
@@ -379,9 +379,14 @@ def handle(self, *args, **options):
         # Should clean_pyc for (clean) reinstall purposes
         # call_command("clean_pyc", interactive=False, verbosity=options.get("verbosity"), path=os.path.join(settings.PROJECT_PATH, ".."))
 
-        # Migrate the database
-        call_command("syncdb", interactive=False, verbosity=options.get("verbosity"))
-        call_command("migrate", merge=True, verbosity=options.get("verbosity"))
+        # If a db template exists, copy it instead of creating a new one, since migrations take a long time.
+        if settings.DB_TEMPLATE_FILE and not os.path.isfile(settings.DEFAULT_DATABASE_PATH):
+            print("Copying database file from {0} to {1}".format(settings.DB_TEMPLATE_FILE, settings.DEFAULT_DATABASE_PATH))
+            shutil.copy(settings.DB_TEMPLATE_FILE, settings.DEFAULT_DATABASE_PATH)
+        else:
+            print("Baking a fresh database from scratch.")
+            call_command("syncdb", interactive=False, verbosity=options.get("verbosity"))
+            call_command("migrate", merge=True, verbosity=options.get("verbosity"))
```

Comment:
I think existing databases should always have `migrate` run?

```
    elif os.path.isfile(settings.DEFAULT_DATABASE_PATH):
        try:
            call_command("migrate", merge=True, verbosity=options.get("verbosity"))
        except Exception as e:
            logger.error("Encountered exception while migrating {}, exception was: {}".format(settings.DEFAULT_DATABASE_PATH, e))
            logger.error("Consider deleting your database {} and running kalite manage setup again.".format(settings.DEFAULT_DATABASE_PATH))
```

Not sure if it's robust enough... but I added the exception handling in case someone kills their database creation mid-time or in some other way end up with a corrupted DB.